plugins/modules/system/puppet.py

```diff
@@ -246,7 +245,8 @@ def main():
         if p['puppetmaster']:
             cmd += " --server %s" % shlex_quote(p['puppetmaster'])
         if p['show_diff']:
-            cmd += " --show_diff"
+            module.deprecate("The 'show_diff' parameter is deprecated, use module diff instead ",
```

Comment:
No need for this, the `removed_in_version` in argspec will already print that message.

Reply:
but do I keep the previous command `cmd += "--show-diff"`?

Reply:
Yes, that should definitely stay.